docs/pages/dropdown.js

```diff
@@ -1,44 +1,22 @@
 // @flow strict
 import { type Node } from 'react';
 import { Dropdown } from 'gestalt';
-import PropTable from '../components/PropTable.js';
 import PageHeader from '../components/PageHeader.js';
 import MainSection from '../components/MainSection.js';
-import docgen, { type DocGen } from '../components/docgen.js';
+import { multipledocgen, type DocGen } from '../components/docgen.js';
 import Page from '../components/Page.js';
+import GeneratedPropTable from '../components/GeneratedPropTable.js';
 
-const commonDropdownItemProps = [
-  {
-    name: 'badgeText',
-    type: 'string',
-    description:
-      "When supplied, will display a [Badge](/badge) next to the item's label. See the [Badges](#Badges) variant to learn more.",
-  },
-  {
-    name: 'dataTestId',
-    type: 'string',
-    description: 'When supplied, will add a data-test-id prop to the dom element.',
-  },
-  {
-    name: 'children',
-    type: 'React.Node',
-    description:
-      'If needed, users can supply custom content to each Dropdown Item. This can be useful when extra functionality is needed beyond a basic Link. See the [Custom item content](#Custom-item-content) variant to learn more.',
-  },
-  {
-    name: 'option',
-    type: '{| label: string, value: string, subtext?: string |}',
-    required: true,
-    description: 'Object detailing the label, value, and optional subtext for this item.',
-  },
-];
-
-export default function DropdownPage({ generatedDocGen }: {| generatedDocGen: DocGen |}): Node {
+export default function DropdownPage({
+  generatedDocGen,
+}: {|
+  generatedDocGen: Array<DocGen>,
+|}): Node {
   return (
     <Page title="Dropdown">
       <PageHeader
         name="Dropdown"
-        description={generatedDocGen?.description}
+        description={generatedDocGen[0]?.description}
```

Comment:
Using array indices is pretty brittle. Instead of an array, can we use an object/map to allow for access by component name?

Reply:
It looks like the component name is available…after awaiting the Promise.all, it looks like you could use reduce to create a map keyed on the component name?

Reply:
Refactored it to a map, however Promise.all must return iterable and reducer method didn't seem to work  well  with Promise.all.